core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/SdkExecutionAttribute.java

```diff
@@ -115,6 +144,78 @@ public class SdkExecutionAttribute {
     public static final ExecutionAttribute<ChecksumValidation> HTTP_RESPONSE_CHECKSUM_VALIDATION = new ExecutionAttribute<>(
         "HttpResponseChecksumValidation");
 
+    private static final ImmutableMap<Algorithm, ChecksumAlgorithm> CHECKSUM_ALGORITHM_MAP = ImmutableMap.of(
+        Algorithm.SHA256, SHA256,
+        Algorithm.SHA1, SHA1,
+        Algorithm.CRC32, CRC32,
+        Algorithm.CRC32C, CRC32C
+    );
+
+    private static final ImmutableMap<ChecksumAlgorithm, Algorithm> ALGORITHM_MAP = ImmutableMap.of(
+        SHA256, Algorithm.SHA256,
+        SHA1, Algorithm.SHA1,
+        CRC32, Algorithm.CRC32,
+        CRC32C, Algorithm.CRC32C
+    );
+
     protected SdkExecutionAttribute() {
     }
+
+    private static <T extends Identity> ChecksumSpecs signerChecksumReadMapping(SelectedAuthScheme<T> authScheme) {
+        if (authScheme == null) {
+            return null;
+        }
+        ChecksumAlgorithm checksumAlgorithm =
+            authScheme.authSchemeOption().signerProperty(AwsV4FamilyHttpSigner.CHECKSUM_ALGORITHM);
+
+        if (checksumAlgorithm == null) {
+            return null;
+        }
+
+        return ChecksumSpecs.builder()
+                            .headerName("x-amz-checksum-" + checksumAlgorithm.algorithmId().toLowerCase(Locale.US))
+                            .algorithm(ALGORITHM_MAP.getOrDefault(checksumAlgorithm, null))
+                            .build();
+    }
+
+    private static <T extends Identity> SelectedAuthScheme<?> signerChecksumWriteMapping(SelectedAuthScheme<T> authScheme,
+                                                                                         ChecksumSpecs checksumSpecs) {
+        ChecksumAlgorithm checksumAlgorithm =
+            checksumSpecs == null ? null
+                                  : CHECKSUM_ALGORITHM_MAP.getOrDefault(checksumSpecs.algorithm(), null);
+
+        if (authScheme == null) {
+            // This is an unusual use-case.
```

Comment:
does it make sense to log something here?

Reply:
Probably not. It would be very rare that this branch is hit.